packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -0,0 +1,33 @@
+import type { Nuxt } from '@nuxt/schema';
+import { sentryVitePlugin } from '@sentry/vite-plugin';
+import type { SentryNuxtModuleOptions } from '../common/types';
+
+/**
+ *  Setup source maps for Sentry inside the Nuxt module during build time.
+ */
+export function setupSourceMaps(moduleOptions: SentryNuxtModuleOptions, nuxt: Nuxt): void {
+  nuxt.hook('vite:extendConfig', async (viteInlineConfig, _env) => {
+    const sourceMapsUploadOptions = moduleOptions.sourceMapsUploadOptions || {};
+
+    if ((sourceMapsUploadOptions.enabled ?? true) && viteInlineConfig.mode !== 'development') {
+      const sentryPlugins = sentryVitePlugin({
+        org: sourceMapsUploadOptions.org ?? process.env.SENTRY_ORG,
+        project: sourceMapsUploadOptions.project ?? process.env.SENTRY_PROJECT,
+        authToken: sourceMapsUploadOptions.authToken ?? process.env.SENTRY_AUTH_TOKEN,
+        telemetry: sourceMapsUploadOptions.telemetry ?? true,
+        _metaOptions: {
+          telemetry: {
+            metaFramework: 'nuxt',
+          },
+        },
+        debug: moduleOptions.debug ?? false,
+      });
+
+      viteInlineConfig.plugins = viteInlineConfig.plugins || [];
+      viteInlineConfig.plugins.push(...sentryPlugins);
+
+      viteInlineConfig.build = viteInlineConfig.build || {};
+      viteInlineConfig.build.sourcemap = true;
```

Comment:
m: If `viteInlineConfig.build.sourcemap` wasn't `true` before we set it, I'd recommend to log out a warning that we're changing their build config to enable emitting source maps. It's fine to do that but users are worried that their source maps then get deployed to prod. 
I'd recommend we expose the `filesToDeleteAfterUpload` option from the plugin and hint them in the warning that they should set it if they want to delete source maps afterwards again.